flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/module/ModuleManager.java

```diff
@@ -173,16 +174,21 @@ public void useModules(String... names) {
      * @return an optional of {@link FunctionDefinition}
      */
     public Optional<FunctionDefinition> getFunctionDefinition(String name) {
-        Optional<String> module =
-                usedModules.stream()
-                        .filter(
-                                n ->
-                                        loadedModules.get(n).listFunctions().stream()
-                                                .anyMatch(e -> e.equalsIgnoreCase(name)))
-                        .findFirst();
-        if (module.isPresent()) {
-            LOG.debug("Got FunctionDefinition '{}' from '{}' module.", name, module.get());
-            return loadedModules.get(module.get()).getFunctionDefinition(name);
+        AtomicReference<String> moduleName = new AtomicReference<>();
+        usedModules.stream()
+                .anyMatch(
+                        n -> {
+                            if (loadedModules.get(n).listFunctions().stream()
+                                    .anyMatch(name::equalsIgnoreCase)) {
+                                moduleName.set(n);
+                                return true;
+                            }
+                            return false;
+                        });
+
+        if (moduleName.get() != null) {
+            LOG.debug("Got FunctionDefinition '{}' from '{}' module.", name, moduleName.get());
+            return loadedModules.get(moduleName.get()).getFunctionDefinition(name);
```

Comment:
We don't need use `AtomicReference` here, you can refactor this logic into while loop and return once the current module contains the requried function name.